internal/controller/controllers/agent_controller.go

```diff
@@ -250,6 +252,16 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, origAgent, &h.Host, h.ClusterID, nil, false)
 }
 
+func updateAnnotations(log logrus.FieldLogger, agent *v1beta1.Agent, h *models.Host) bool {
+	updated := false
+	updated = setAgentAnnotation(log, agent, AgentStateAnnotation, swag.StringValue(h.Status))
+	updated = updated || setAgentAnnotation(log, agent, AgentRoleAnnotation, string(h.Role))
+	if h.Inventory != "" {
```

Comment:
Just a minor nit for later, is it crucial to check if `inventory` is empty?
If not, we can make this func nicer to read I think.
I.e.
```
updated := false
updated = setAgentAnnotation(log, agent, AgentStateAnnotation, swag.StringValue(h.Status)
updated = setAgentAnnotation(log, agent, AgentRoleAnnotation, string(h.Role)) || updated
updated = setAgentAnnotation(log, agent, AgentInventoryAnnotation, h.Inventory) || updated
```

Reply:
Great suggestion! Will work that into a new PR. Thanks Daniel!!

Reply:
Small PR with changes: https://github.com/openshift/assisted-service/pull/6967